File: code/jjuny-won/week6/boj_12865.py
import sys

N,K = map(int,input().split())
items = []
for _ in range(N):
    w,v = map(int,input().split())
    items.append((w,v))
    

dp = [0 for _ in range(K+1)] #최대 가치를 구하기 위한 일차원 배열


for item in items:
    w,v = item
    for i in range(K,w-1,-1):
        dp[i] = max(dp[i],dp[i-w]+v) 
        #물건을 추가하기전 최대 가치와, 물건을 추가하고 난 후 최대 가치 비교해서 가치 업데이트
print(dp[-1])

File: code/HoChanny/week6/boj_12865.js
const input = require('fs').readFileSync('dev/stdin').toString().trim().split('\n');

const k = +input.shift().split(' ')[1];
const things = input.map(e => e.split(' ').map(v => +v));
const dp = new Array(k + 1).fill(0);

things.forEach((e) => {
    let [w, v] = e;
    for(let i = k; i >= w; i--){
        dp[i] = Math.max(dp[i], dp[i - w] + v);
    }
})

console.log(dp[k]);
